src/utils/daos_hdlr.c

```diff
@@ -1792,59 +1890,63 @@ fs_copy(struct cmd_args_s *ap,
 	char		*tmp_name = NULL;
 
 	/* Make sure the source exists. */
-	rc = file_lstat(ap, src_file_dfs, src_path, &src_stat);
+	rc = file_lstat(copy_args->ap, copy_args->src_file_dfs, copy_args->src_path, &src_stat);
 	if (rc != 0) {
 		rc = daos_errno2der(rc);
-		DH_PERROR_DER(ap, rc, "Failed to stat '%s'", src_path);
+		DH_PERROR_DER(copy_args->ap, rc, "Failed to stat '%s'", copy_args->src_path);
 		D_GOTO(out, rc);
 	}
 
 	/* If the destination exists and is a directory,
 	 * copy INTO the directory instead of TO it.
 	 */
-	rc = file_lstat(ap, dst_file_dfs, dst_path, &dst_stat);
+	rc = file_lstat(copy_args->ap, copy_args->dst_file_dfs, copy_args->dst_path, &dst_stat);
 	if (rc == 0) {
 		if (S_ISDIR(dst_stat.st_mode)) {
 			copy_into_dst = true;
 		} else if S_ISDIR(src_stat.st_mode) {
 			rc = -DER_INVAL;
-			DH_PERROR_DER(ap, rc, "Destination is not a directory");
+			DH_PERROR_DER(copy_args->ap, rc, "Destination is not a directory");
```

Comment:
```suggestion
			DH_PERROR_DER(copy_args->ap, rc, "Destination exists and is not a directory");
```

Reply:
Updated the error msg. Also put num_dirs, num_files and num_links in a struct. Now it makes more sense. Thank you!